Nodejs/Product/Nodejs/TestFrameworks/mocha/mocha.js

```diff
@@ -80,18 +80,30 @@ function logError() {
 
 function detectMocha(projectFolder) {
   try {
-        var node_modulesFolder = projectFolder;
-        var mochaJsonPath = path.join(node_modulesFolder, 'test', 'mocha.json');
-        if (fs.existsSync(mochaJsonPath)) {
-          var opt = require(mochaJsonPath);
-          if (opt && opt.path) {
-            node_modulesFolder = path.resolve(projectFolder, opt.path);
-          }
+    var node_modulesFolder = projectFolder;
+
+    var nodePath = process.env.NODE_PATH;
+    if (nodePath && nodePath.length > 0) {
+      // get mocha location from NODE_PATH env 
+      var ndex = nodePath.search(new RegExp(/node_modules$/i));
```

Comment:
Also, I believe that NODE_PATH is actually a delimited set of paths much like PATH. This case should be handled as well. The separator is different on windows vs unix too. Here's some more info: https://nodejs.org/api/modules.html#modules_loading_from_the_global_folders